File: LayoutTests/http/tests/security/resources/image-credential-check.php
<?php
if ($_SERVER['PHP_AUTH_USER'] == 'testusername' && $_SERVER['PHP_AUTH_PW'] == 'testpassword') {
    $fp = fopen('black-square.png', 'rb');
    header("Content-Type: image/png");
    header("Content-Length: " . filesize('black-square.png'));
    fpassthru($fp);
    exit;
}
header('HTTP/1.0 401 Unauthorized');
header('WWW-Authenticate: Basic realm="test realm"');
exit;
?>

File: LayoutTests/http/tests/security/resources/image-credential-check.py
#!/usr/bin/env python3

import base64
import os
import sys

credentials = base64.b64decode(os.environ.get('HTTP_AUTHORIZATION', ' Og==').split(' ')[1]).decode().split(':')
username = credentials[0]
password = ':'.join(credentials[1:])

if username == 'testusername' and password == 'testpassword':
    sys.stdout.write('Content-Type: image/png\r\n\r\n')
    sys.stdout.flush()
    with open(os.path.join(os.path.dirname(__file__), 'black-square.png'), 'rb') as file:
        sys.stdout.buffer.write(file.read())
    sys.exit(0)

sys.stdout.write(
    'Content-Type: text/html\r\n'
    'status: 401\r\n'
    'WWW-Authenticate: Basic realm="test realm"\r\n\r\n'
)
